### PR TITLE
Web UI development workshop

### PR DESCRIPTION
**Pull request by Diksha-Verma-Dev**
Web UI development workshop: https://nuxeowiki.atlassian.net/wiki/spaces/NuxEng/pages/3244458028/Web+UI+development+workshop

### CODE REVIEW

File: ui/i18n/messages.json [hunk 1]
```diff
@@ -365,5 +365,7 @@
   "workflowButton.dialog.start": "Start",
   "workflowButton.dialog.heading": "Start Process",
   "workflowButton.dialog.placeholder": "Choose process",
-  "workflowButton.tooltip": "Start Process"
+  "workflowButton.tooltip": "Start Process",
+  "favoritesToggleButton.confirm.add": "Are you sure you want to add this document to Favorites?",
+  "favoritesToggleButton.confirm.remove": "Are you sure you want to remove this document from Favorites?"
```
Comment (Gabez0r) [inline ui/i18n/messages.json]: Usually, we try to keep the keys in the JSON object sorted alphanumerically, so these labels should be closer to the others starting with "f".

File: ui/test/nuxeo-favorites-toggle-button.test.js [hunk 2]
```diff
@@ -48,12 +48,19 @@ suite('nuxeo-favorites-toggle-button', () => {
...
+        window.confirm.restore();
+      }
+    });
+
     test('it should display the document as favorite', () => {
       expect(element.favorite).to.be.true;
     });
 
-    test('toggle should remove document from favorites', async () => {
+    test('toggle should remove document from favorites when clicked ok', async () => {
```
Comment (Gabez0r) [inline ui/test/nuxeo-favorites-toggle-button.test.js:62]: We should also test that it does not remove if the user clicks cancel.

**Review by Gabez0r — COMMENTED** [review]
Looking good!

File: ui/test/nuxeo-favorites-toggle-button.test.js [hunk 3]
```diff
@@ -48,16 +48,30 @@ suite('nuxeo-favorites-toggle-button', () => {
...
       tap(element);
       await waitChanged(element, 'favorite');
       expect(element.favorite).to.be.false;
     });
+
+    test('toggle should not remove document from favorites when clicked Cancel', async () => {
+      // Remove document from favorites by toggling
+
+      sinon.stub(window, 'confirm').returns(false);
+      expect(element.favorite).to.be.true;
```
Comment (Gabez0r) [inline ui/test/nuxeo-favorites-toggle-button.test.js:75]: Didn't we forgot to click the button?

File: ui/test/nuxeo-favorites-toggle-button.test.js [hunk 4]
```diff
@@ -84,13 +98,24 @@ suite('nuxeo-favorites-toggle-button', () => {
...
     });
+    teardown(() => {
+      if (window.confirm.restore) {
+        window.confirm.restore();
+      }
+    });
+
+    test('toggle should not add the document to favorites when clicked on Cancel', async () => {
+      sinon.stub(window, 'confirm').returns(false);
+      expect(element.favorite).to.be.false;
```
Comment (Gabez0r) [inline ui/test/nuxeo-favorites-toggle-button.test.js]: Didn't we forgot to click the button?

**Review by Gabez0r — COMMENTED** [review]
Please see the latest comments in https://github.com/nuxeo/nuxeo-elements/pull/539

File: ui/actions/nuxeo-favorites-toggle-button.js [hunk 5]
```diff
@@ -147,25 +147,29 @@ import '../nuxeo-button-styles.js';
...
         this.$.opAdd.execute().then(() => {
-          this.dispatchEvent(
-            new CustomEvent('added-to-favorites', {
-              composed: true,
-              bubbles: true,
-              detail: { doc: this.document },
-            }),
-          );
-          this._setFavorite(true);
+          if (window.confirm(this.i18n('favoritesToggleButton.confirm.add')) === true) {
```
Comment (Gabez0r) [inline ui/actions/nuxeo-favorites-toggle-button.js]: Please note that we are asking the user to confirm AFTER we already executed the operation, which is not what we want here.

File: ui/i18n/messages.json [hunk 6]
```diff
@@ -242,6 +244,7 @@
   "operationButton.bulk.poll.error": "Error while executing {0}",
   "operationButton.bulk.poll.running": "{0} is running ({1}/{2})",
   "operationButton.bulk.poll.scheduled": "{0} is scheduled. You will be notified when it starts.",
+  "parentInspectorButton.tooltip": "Parent Inspector",
```
Comment (Gabez0r) [inline ui/i18n/messages.json]: It this label is not used in this repository but only in the nuxeo web ui repository, then that's where it should live.

**Review by richardsd — COMMENTED** [review]
Looking good. Please relaunch the https://github.com/nuxeo/nuxeo-elements/runs/7108971407?check_suite_focus=true Web UI PR check since it was failing due to a random failure on Web UI unit tests.

**Pull request closed without merging**